### PR TITLE
Trigger plotly_relayout event when camera is reset to default or saved (Solves #FI29)

### PR DESCRIPTION
**Pull request by monfera**
Fixes issue #FI29. The added test cases aren't part of CircleCI as the test file is excluded, but they pass when run locally with fdescribe.

### CODE REVIEW

**Comment by monfera** [comment]
@etpinard @bpostlethwaite I believe these changes address FI-29.

File: src/plots/gl3d/scene.js [hunk 1]
```diff
@@ -559,11 +559,13 @@ proto.destroy = function() {
...
     // as in Gl3d.layoutAttributes
-    this.glplot.camera.lookAt(
+    var lookAtInput = [
         [1.25, 1.25, 1.25],
         [0, 0, 0],
         [0, 0, 1]
-    );
+    ];
+    this.glplot.camera.lookAt.apply(this, lookAtInput);
+    this.graphDiv.emit('plotly_relayout', lookAtInput);
```
Comment (etpinard) [inline src/plots/gl3d/scene.js]: We could do even better passing the corresponding [camera attribute](https://github.com/plotly/plotly.js/blob/3180b1254668b8791e20766f77707d5601d69a49/src/plots/gl3d/layout/layout_attributes.js#L41-L66) update object e.g:

```js
this.graphDiv.emit('plotly_relayout', {
  scene: {
    eye: { x: 1.25, y: 1.25, z: 1.25 },
    center: { x: 0, y: 0, z: 0 },
    up: { x: 0, y: 0, z: 1 }
  }
});
```

Comment (monfera) [inline src/plots/gl3d/scene.js]: @etpinard I'm glad to refactor for it; I'll extract the object-to-array logic into a separate function so we can be DRY among these two `emit` places.

File: src/plots/gl3d/scene.js [hunk 2]
```diff
@@ -586,11 +588,13 @@ proto.setCamera = function setCamera(cameraData) {
     var up = cameraData.up;
     var center = cameraData.center;
     var eye = cameraData.eye;
-    this.glplot.camera.lookAt(
+    var lookAtInput = [
```
Comment (etpinard) [inline src/plots/gl3d/scene.js]: similarly here.

File: src/plots/gl3d/scene.js (hunk 1 above)
Comment (etpinard) [inline src/plots/gl3d/scene.js]: :+1:

Comment (etpinard) [inline src/plots/gl3d/scene.js]: Oh. To handle the mulit-scene case :arrow_double_up: should in fact be:

```js
var update = {}l

update[this.id] = {
    eye: { x: 1.25, y: 1.25, z: 1.25 },
    center: { x: 0, y: 0, z: 0 },
    up: { x: 0, y: 0, z: 1 }
  };

this.graphDiv.emit('plotly_relayout', update);
```

**Comment by etpinard** [comment]
Ok. I'm *ok* with this.

We should spend some time rethinking how `scene.camera` interacts with `Plotly.relayout` so that e.g. the mode bar buttons could simply call `Plotly.relayout` with the correct update object without having to use an internal `Scene` method.

That will be for another day (possibly before `v2.0.0`) though.

File: test/jasmine/tests/gl_plot_interact_test.js [hunk 3]
```diff
@@ -353,7 +369,18 @@ describe('Test gl plot interactions', function() {
                         .toBeCloseToArray([1.25, 1.25, 1.25], 4);
 
                     selectButton(modeBar, 'resetCameraLastSave3d').click();
+
                     setTimeout(function() {
+
+                        expect(relayoutCallback).toHaveBeenCalled(); // initiator: resetCameraLastSave3d
+                        expect(relayoutCallback).toHaveBeenCalledWith({
+                            scene: {
+                                eye: { x: 1.25, y: 1.25, z: 1.25 },
```
Comment (etpinard) [inline test/jasmine/tests/gl_plot_interact_test.js]: This assertion is misleading. Adding

```js
console.log(getOrbitCamera(cameraData))
```

above this [line](https://github.com/monfera/plotly.js/blob/29-emit-relayout-events-on-3d-camera-reset/src/plots/gl3d/scene.js#L599) in `Scene.prototype.setCamera` logs

```js
[[0.1, 0.1, 1], [0, 0, 0], [0, 0, 1]]
```

for the first scene, and

```js
[[2.5, 2.5, 2.5], [0, 0, 0], [0, 0, 1]]
```

for the second scene.

I think the jasmine spy, holds on to the arguments of the first `relayoutCallback` call and isn't updated for subsequent calls.

Moreover, we should make it clear that `plotly_relayout` is triggered twice (one per each scene) when clicking on these buttons.

File: src/plots/gl3d/scene.js [hunk 4]
```diff
@@ -559,11 +559,12 @@ proto.destroy = function() {
 // for reset camera button in mode bar
 proto.setCameraToDefault = function setCameraToDefault() {
     // as in Gl3d.layoutAttributes
-    this.glplot.camera.lookAt(
-        [1.25, 1.25, 1.25],
-        [0, 0, 0],
-        [0, 0, 1]
-    );
+
+    this.setCamera({
```
Comment (etpinard) [inline src/plots/gl3d/scene.js]: nice clean up.

File: test/jasmine/tests/gl_plot_interact_test.js (hunk 3 above)
Comment (monfera) [inline test/jasmine/tests/gl_plot_interact_test.js]: Thanks @etpinard, I'll try to capture the multi-scene aspect.

File: test/jasmine/tests/gl_plot_interact_test.js [hunk 5]
```diff
@@ -350,14 +350,22 @@ describe('Test gl plot interactions', function() {
...
                         }
                     });
+                    expect(relayoutCallback).toHaveBeenCalledWith({
+                        scene2: {
+                            center: { x: 0, y: 0, z: 0 },
+                            eye: { x: 1.25, y: 1.25, z: 1.25 },
+                            up: { x: 0, y: 0, z: 1 }
+                        }
+                    });
+                    relayoutCallback.calls.reset();
```
Comment (etpinard) [inline test/jasmine/tests/gl_plot_interact_test.js]: nice move.

**Comment by etpinard** [comment]
@monfera Great way to get introduced with our gl3d code. Nicely done :beers:

**Pull request merged**